ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -405,7 +405,14 @@ func (r *RayDashboardClient) StopJob(jobName string, log *logr.Logger) (err erro
 	}
 
 	if !jobStopResp.Stopped {
-		return fmt.Errorf("failed to stopped job: %v", jobName)
+		jobInfo, err := r.GetJobInfo(jobName)
+		if err != nil {
+			return err
+		}
+		// StopJob only returns an error when JobStatus is not in terminated states (STOPPED / SUCCEEDED / FAILED)
+		if (jobInfo.JobStatus == rayv1alpha1.JobStatusPending) || (jobInfo.JobStatus == rayv1alpha1.JobStatusRunning) {
```

Comment:
Since we might change the states over time, I would suggest 

- Defining the terminated states in a constant somewhere.
- Have this condition check if we're not terminated, using that constant.

Reply:
Good idea! I will update it tomorrow.

Reply:
Updated. I decided not to replace `isJobSucceedOrFailed` and `isJobPendingOrRunning` with `IsJobTerminal` in this PR because current RayJob controller does not handle jobs with `STOPPED` statuses explicitly. I will open an issue to track the progress.

https://github.com/ray-project/kuberay/blob/792af2a6e8411ba10f9fe5962142cc6385ee549e/ray-operator/controllers/ray/rayjob_controller.go#L239-L253